src/uct/api/uct.h

```diff
@@ -1655,6 +1669,54 @@ UCT_INLINE_API ucs_status_t uct_ep_am_zcopy(uct_ep_h ep, uint8_t id,
                                       flags, comp);
 }
 
+
+/**
+ * @ingroup UCT_AM
+ * @brief Register a memory layout for transfers over this endpoint.
+ *
+ * The input data in @a iov array of @ref ::uct_iov_t structures describes a
+ * memory layout to be registered with this endpoint. Unless the function returns
+ * with an error, subsequent calls to send or recieve operations with this endpoint
+ * may use the resulting handle in @a memh_p to operate on data with that layout.
+ * This function requires @ref UCT_IFACE_FLAG_REG . Once the registration
+ * is complete - any endpoint in the same domain can use the resulting memory handle.
+ *
+ * The layout is composed of one or more @ref ::uct_iov_t objects. For example,
+ * a simple contiguous buffer would be a single object containing a pointer to
+ * the buffer and its length (count is 1, and the rest of the fields are zero).
+ * For a complex memory layout, such as two interleaved vectors, the function
+ * should accept two objects, one for each vector, each set with the stride between
+ * two items, the length of an item, their amount and the interleaving ratio:
+ * how many of items from a vector to put before moving to the next vector (for
+ * a pattern of AABAABAAB... the interleaving ratio is 2 for vector A and 1 for B).
+ *
+ * The user must destroy the handle by calling @ref uct_md_mem_free with @a md_p ,
+ * which can be done regardless of completion of the registration (this call must
+ * return, but there is no need to wait for or even specify a completion callback).
+ *
+ * @param [in]  ep           Destination endpoint handle.
+ * @param [in]  iov          Points to an array of @ref ::uct_iov_t structures.
+ *                           The @a iov pointer must be valid address of an array
+ *                           of @ref ::uct_iov_t structures. A particular structure
+ *                           pointer must be valid address. NULL terminated pointer
+ *                           is not required.
+ * @param [in]  iovcnt       Size of the @a iov data @ref ::uct_iov_t structures
+ *                           array. If @a iovcnt is zero, the data is considered empty.
+ *                           @a iovcnt is limited by @ref uct_iface_attr_cap_am_max_iov
+ *                           "uct_iface_attr::cap::mem::max_iov"
+ * @param [out] md_p         Filled with the memory domain handle, for destruction.
```

Comment:
why is memory domain an output parameter?
why do we even need it, we already have endpoint parameter which is on a specific memory domain?